src/WorldStorage/NBTChunkSerializer.cpp

```diff
@@ -555,6 +556,33 @@ void cNBTChunkSerializer::AddMonsterEntity(cMonster * a_Monster)
 		m_Writer.AddByte("CanPickUpLoot", (a_Monster->CanPickUpLoot())? 1 : 0);
 		m_Writer.AddString("CustomName", a_Monster->GetCustomName());
 		m_Writer.AddByte("CustomNameVisible", static_cast<Byte>(a_Monster->IsCustomNameAlwaysVisible()));
+
+		// Mob was leashed
+		if (a_Monster->IsLeashed() || (a_Monster->GetLeashToPos() != nullptr))
+		{
+			m_Writer.AddByte("Leashed", 1);
+
+			const Vector3d * LeashedToPos = nullptr;
+
+			if (a_Monster->GetLeashToPos() != nullptr)
+			{
+				LeashedToPos = a_Monster->GetLeashToPos();
+			}
+			else if (a_Monster->GetLeashedTo()->IsLeashKnot())
```

Comment:
No, because the outer _if_ ensures that one of them is not null. 
a_Monster->IsLeashed() it's the same as a_Monster->GetLeashedTo() != nullptr